test/jdk/javax/swing/plaf/gtk/JTableCtrlShiftRightLeftKeyTest.java

```diff
@@ -0,0 +1,138 @@
+        UIManager.setLookAndFeel("com.sun.java.swing.plaf.gtk.GTKLookAndFeel");
+        robot = new Robot();
+        robot.setAutoDelay(50);
+        try {
+            SwingUtilities.invokeAndWait(JTableCtrlShiftRightLeftKeyTest::createAndShowUI);
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            SwingUtilities.invokeAndWait(() -> {
+                tableLoc = table.getLocationOnScreen();
+                cellRect = table.getCellRect(0, SELECTED_COLUMN, true);
+            });
+
+            robot.mouseMove(tableLoc.x + cellRect.x + cellRect.width / 2,
+                    tableLoc.y + cellRect.y + cellRect.height / 2);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
+            robot.delay(100);
+
+            testCtrlShift(KeyEvent.VK_RIGHT, SELECTED_COLUMN, table.getColumnCount()-1);
+
+            robot.waitForIdle();
+            robot.delay(100);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
+            robot.delay(100);
+
+            testCtrlShift(KeyEvent.VK_LEFT, 0, SELECTED_COLUMN);
+            robot.waitForIdle();
+            robot.delay(100);
+            System.out.println("Test Passed!");
+
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    private static void testCtrlShift(int keySelected, int startCellCheck,
+                                      int endCellCheck) throws Exception {
+        robot.keyPress(KeyEvent.VK_SHIFT);
+        robot.keyPress(KeyEvent.VK_CONTROL);
+        robot.keyPress(keySelected);
+        robot.keyRelease(keySelected);
+        robot.keyRelease(KeyEvent.VK_SHIFT);
+        robot.keyRelease(KeyEvent.VK_CONTROL);
+        robot.waitForIdle();
+        robot.delay(100);
+
+        SwingUtilities.invokeAndWait(() -> {
+            selectedColumnAfterKeyPress = table.getSelectedColumns();
+        });
+
+        if (selectedColumnAfterKeyPress[0] != startCellCheck ||
+                selectedColumnAfterKeyPress[selectedColumnAfterKeyPress.length-1] !=
+                        endCellCheck) {
+            System.out.println("Selected Columns: ");
+            for (int columnAfterTabPress : selectedColumnAfterKeyPress) {
+                System.out.println(columnAfterTabPress);
+            }
+            String key = (keySelected == KeyEvent.VK_RIGHT)? "RIGHT" : "LEFT";
+            String failureMsg = "Test Failure. Failed to select cells for Ctrl" +
+                    " Shift "+key+" selection";
+            throw new RuntimeException(failureMsg);
+        }
+    }
+
+    private static void createAndShowUI() {
+        frame = new JFrame("Test JTable Tab Press");
```

Comment:
Title should be changed to something like `Test JTable Ctrl + Shift + Left / Right Key Press`